From 75376eeac94b09a3e19f9f221b493f089da49a96 Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Mon, 4 Mar 2024 02:18:01 +0000
Subject: [PATCH] pfe: net-if: replace sa_data with latest sa_data_min to fix
 build failure

The kernel commit 4d3b2bd995ed ("net: dev: Convert sa_data to flexible array
in struct sockaddr") updated the sa_data member of sockaddr struct, so
update pfe related codes to keep align with kernel codes, or else there
will be following build errors:

  .../sw/linux-pfeng/pfeng-netif.c:1308:57: error: invalid application of 'sizeof' to incomplete type 'char[]'
  |  1308 |                         memset(&saddr.sa_data, 0, sizeof(saddr.sa_data));
  |       |                                                         ^

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 sw/linux-pfeng/pfeng-netif.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sw/linux-pfeng/pfeng-netif.c b/sw/linux-pfeng/pfeng-netif.c
index 7872b8b..5d5d2a7 100644
--- a/sw/linux-pfeng/pfeng-netif.c
+++ b/sw/linux-pfeng/pfeng-netif.c
@@ -1305,9 +1305,9 @@ static int pfeng_netif_logif_init_second_stage(struct pfeng_netif *netif)
 		if (is_valid_ether_addr(netif->cfg->macaddr))
 			memcpy(&saddr.sa_data, netif->cfg->macaddr, ARRAY_SIZE(netif->cfg->macaddr));
 		else
-			memset(&saddr.sa_data, 0, sizeof(saddr.sa_data));
+			memset(&saddr.sa_data, 0, sizeof(saddr.sa_data_min));
 	} else
-		memcpy(&saddr.sa_data, netdev->dev_addr, sizeof(saddr.sa_data));
+		memcpy(&saddr.sa_data, netdev->dev_addr, sizeof(saddr.sa_data_min));
 
 	pfeng_netif_set_mac_address(netdev, (void *)&saddr);
 
-- 
2.34.1

